### PR TITLE
SDK-302 [Node.js SDK] Allow oAuth2 token Storage initialization

### PR DESCRIPTION
**Pull request by artemgolozubov**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-302](https://incountry.atlassian.net/browse/SDK-302)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [x] Please put the target release vehicle this needs to be bundled with.
- [x] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ type StorageOptions = {
       default: string;
       [key: string]: string;
     };
+    token?: string;         // Used when OAuth token is already received by SDK user. Mutually exclusive with clientId, clientSecret, authEndpoints
```
Comment (RommeroInCountry) [inline README.md:50]: Used when OAuth token is already acquired prior to Storage initialization.

File: README.md [hunk 2]
```diff
@@ -126,6 +127,21 @@ const storage = new Storage({
 ```
 
 
+The SDK also allows to use already received OAuth tokens if needed. In this mode SDK user is responsible for OAuth token receiving and renewal. Use this option only at your own risk.
```
Comment (RommeroInCountry) [inline README.md:130]: "... to use previously acquired oAuth tokens if needed. In this mode SDK is not responsible for oAuth token renewal and it should be done by SDK user himself"

Let's also drop "risk" sentence :)

File: README.md [hunk 3]
```diff
@@ -126,6 +127,21 @@ const storage = new Storage({
...
+
+Below you can find the example of how to specify OAuth token while creating a storage instance:
```
Comment (RommeroInCountry) [inline README.md:132]: storage -> Storage

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

File: src/validation/storage-options.ts [hunk 4]
```diff
@@ -49,11 +56,27 @@ const OAuthEndpointsIO: t.Type<OAuthEndpoints> = new t.Type(
...
   clientSecret: t.string,
   authEndpoints: OAuthEndpointsIO,
-}, 'OAuthOptions');
+}, 'OAuthCredentials'));
+
+const OAuthTokenIO: t.Type<OAuthToken> = exact(t.type({
+  token: t.string,
+}, 'OAuthToken'));
+
+const OAuthOptionsIO: t.Type<OAuthOptions> = new t.Type(
```
Comment (igor-alekseev-dev) [inline src/validation/storage-options.ts:69]: Maybe we can simplify this to `t.union([ A, B ])` ?

File: src/validation/storage-options.ts [hunk 5]
```diff
@@ -49,11 +56,27 @@ const OAuthEndpointsIO: t.Type<OAuthEndpoints> = new t.Type(
   Object,
 );
 
-const OAuthOptionsIO: t.Type<OAuthOptions> = t.partial({
+const OAuthCredentialsIO: t.Type<OAuthCredentials> = exact(t.partial({
   clientId: t.string,
   clientSecret: t.string,
```
Comment (igor-alekseev-dev) [inline src/validation/storage-options.ts:61]: Are `clientId` and `clientSecret` really optional within OAuth credentials?

**Review by igor-alekseev-dev — APPROVED** [review]

File: src/validation/storage-options.ts (hunk 5 above)
Comment (artemgolozubov) [reply src/validation/storage-options.ts:61]: Yeah, 
```
{
  ...,
  oauth: {},
}
```
is a valid config. Client id + secret could be specified in environment variables)

**Review by RommeroInCountry — APPROVED** [review]

**Review by igor-alekseev-dev — APPROVED** [review]

**Pull request merged**